src/utils.ts

```diff
@@ -699,38 +633,62 @@ export function estimateCurrentPrice(order: Order, secondsToBacktrack = 30, shou
   return shouldRoundUp ? exactPrice.ceil() : exactPrice
 }
 
+/**
+ * Wrapper function for getting generic Wyvern assets from OpenSea assets
+ * @param schema Wyvern schema for the asset
+ * @param asset The fungible or nonfungible asset to format
+ */
+export function getWyvernAsset(
+    schema: Schema<WyvernAsset>,
+    asset: Asset,
+    quantity = new BigNumber(1)
+  ) {
+  if (SCHEMA_NAME_TO_ASSET_CONTRACT_TYPE[schema.name as WyvernSchemaName] == AssetContractType.NonFungible) {
```

Comment:
All these type casting is happening because `Schema` doesn't explicitly express polymorphism on `.name`. I think we should rethink the typing.

Reply:
it's because the internal wyvern-schemas lib doesn't have an enum for the different schema names available. opensea-js provides that. one day we should merge the two repos